src/library_postprocessing.rs

```diff
@@ -373,4 +376,52 @@ impl Library {
             }
         }
     }
+
+    fn mark_final_types(&mut self, config: &Config) {
+        // Here we mark all class types as final types if configured so in the config or
+        // otherwise if there is no public class struct for the type.
+        //
+        // Final types can't have any subclasses and we handle them slightly different
+        // for that reason.
+        let mut actions: Vec<TypeId> = Vec::new();
+
+        for (ns_id, ns) in self.namespaces.iter().enumerate() {
+            for (id, type_) in ns.types.iter().enumerate() {
+                let type_ = type_.as_ref().unwrap(); //Always contains something
+
+                match *type_ {
+                    Type::Class(ref klass) => {
+                        let tid = TypeId {
+                            ns_id: ns_id as u16,
+                            id: id as u32,
+                        };
+
+                        let full_name = tid.full_name(self);
+                        let obj = config.objects.get(&*full_name);
+
+                        if let Some(GObject { final_type: Some(final_type), ..}) = obj {
+                            // The config might also be used to override a type that is wrongly
+                            // detected as final type otherwise
+                            if *final_type {
+                                actions.push(tid);
+                            }
+                        } else if klass.c_class_type.is_none() {
+                            actions.push(tid);
```

Comment:
This case IMHO needed comment too

Reply:
There's a comment a few lines above that is explaining the whole code, I would just repeat the same thing here otherwise :)

Reply:
At the top of the function

Reply:
Ok, misunderstand that too.